utility/src/test/java/org/oppia/android/util/profile/ProfileNameValidatorTest.kt

```diff
@@ -0,0 +1,71 @@
+package org.oppia.android.util.profile
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Test
+import javax.inject.Inject
+
+class ProfileNameValidatorTest {
+  @Inject
+  lateinit var profileNameValidator: ProfileNameValidator
+
+  @Before
+  fun setup() {
+    profileNameValidator = ProfileNameValidator()
+  }
+
+  @Test
+  fun testIsNameValid_nameWithSpaces_returnsFalse() {
+    val nameWithSpaces = "Ben Henning"
+    assertThat(profileNameValidator.isNameValid(nameWithSpaces)).isFalse()
+  }
+
+  @Test
+  fun testIsNameValid_nameWithNumber_returnsFalse() {
+    val nameWithSpaces = "Jishnu7"
+    assertThat(profileNameValidator.isNameValid(nameWithSpaces)).isFalse()
+  }
+
+  @Test
+  fun testIsNameValid_nameWithDisallowedSymbol_returnsFalse() {
+    val namesWithSymbols =
+      listOf<String>("Ben#Henning", "Rajay@T", "जिष्णु**", "جيشنو^&&", "Jishnu_")
+    namesWithSymbols.forEach {
+      assertThat(profileNameValidator.isNameValid(it)).isFalse()
+    }
+  }
+
+  @Test
+  fun testIsNameValid_nameWithAllowedSymbols_returnsTrue() {
+    val namesWithAllowedSymbol = listOf<String>("Ben-Henning", "Rajat.T", "G'Jishnu")
+    namesWithAllowedSymbol.forEach {
+      assertThat(profileNameValidator.isNameValid(it)).isTrue()
+    }
+  }
+
+  @Test
+  fun testIsNameValid_nameWithRepeatedAllowedSymbols_returnsFalse() {
```

Comment:
I think this will be a good candidate for parameterized tests, though that'll need to wait for my chain of math expressions PRs to be merged (which might take another week or so). Are you okay with waiting for that?

Reply:
Sure @BenHenning . Is it ready yet?